test/distributed/test_c10d_gloo.py

```diff
@@ -465,6 +465,20 @@ def _test_allreduce_basics(self, fn):
             result[0],
         )
 
+        # Single input tests
+        tensor = (torch.rand(20000, 1, dtype=torch.float32) * 2 - 1) * 65504 / self.world_size
```

Comment:
the np.uniform is a better test than this since it tests every possible bit combination where this only tests the valid fp32 ones (no nan, inf etc)

Reply:
sure, let me use the numpy one.